### ISSUE TITLE
Product Gallery: Consider implementing pseudo infinite loop effect

### ISSUE THREAD

**Issue by danieldudzic** [woothemes/woocommerce#43255]
### Describe the solution you'd like

Currently, when you scroll to the end of available product thumbnails the next button doesn't do anything.

AFAIK it's not possible to utilize a true infinite loop effect with a pure CSS slider like the Product Gallery block utilizes, but we should be able to scroll to the first thumbnail, like here: https://codepen.io/alvarotrigo/pen/bGqgqeK

**Comment by danieldudzic** [comment]
@jarekmorawski What do you think?

**Comment by jarekmorawski** [comment]
Yes, please! I thought we already had this behavior. It's a pretty common pattern. One note, though: let's not have the whole gallery scroll through when the user reaches the last image or clicks the back arrow on the first one. Some products can have 10s of images, so it'd take seconds to scroll through the gallery, not to mention the visual noise it'd cause.

Instead, I'd prefer us to use a linear animation and treat the last/first image as the next image in the gallery. Here's an example: https://codepen.io/frontendmax/pen/abbBzpq

**Comment by jarekmorawski** [comment]
I assume this also means that we'll no longer need the disabled state for the arrow buttons. As for settings, I don't think we need to make the loop toggleable in the block inspector yet. We can revisit this based on user feedback.

**Comment by danieldudzic** [comment]
Thank you for your input, @jarekmorawski.

I specifically suggested the _pseudo_ infinite scrolling https://codepen.io/alvarotrigo/pen/bGqgqeK (which you mentioned is not your preference) as this behavior should be fairly easy to implement (at least that's my assumption) with our pure CSS (`scroll-snap-type`) slider. The example I linked to uses the same `scroll-snap-type` sliding mechanism.

Implementing a true infinite scroll, similar to the one found at https://codepen.io/frontendmax/pen/abbBzpq, will most likely require additional JS code. This somewhat defeats the purpose of relying on a pure CSS slider (trying to keep things lightweight and avoiding reliance on any existing JavaScript libraries/code for scrolling).

Considering these constraints, is a _pseudo_ infinite loop effect a deal-breaker?

**Comment by jarekmorawski** [comment]
It is not. Let's do a pseudo effect first and revisit the behavior later.

**Comment by kmanijak** [comment]
Headsup I'm adding buttons disabling on the first and last image: https://github.com/woocommerce/woocommerce/pull/53318. It's a temporary measure until we added pseudo infinite loop.

**Comment by danielwrobert** [comment]
I’m wondering if this is still needed, considering the improvement put in place in #53318 by @kmanijak? Is there an added benefit to having the infinite scroll, as opposed to the visual indication that the user has reached the beginning or end of the slideshow?

Both patterns seem relatively common so I see value in either direction. On one hand, if you have a very long image list, you can go to the beginning more easily. On the other hand it feels a bit more intuitive (at least to me) as you’re scrolling through to know when you’ve reached the end. 

I’ll defer to @j111q (and @jarekmorawski, since you were involved in the conversations on this Issue) for the final call. My main point in raising the questions is that we have one solution in place and there would be a bit more technical overhead in implementing the infinite scroll approach. I want to make sure the difference in experience is worth that added effort before we take that route.

**Comment by roykho** [comment]
I am currently looking into using Swiperjs to make this work. The benefit to using Swiperjs is that it allows for many different configurations and also solves the issue with touch support. Currently we don't have any touch support which is important. Though we don't actually have a task to work on touch support for this cycle, adding Swiperjs would solve both issues.

But let me know if we think this is too much effort for now and we can pivot.

**Comment by kmanijak** [comment]
Not planned at the moment. We'll create new issue if necessary. I'm closing this one to keep the backlog clean.
